File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/UserService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbUser;

import java.util.List;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface UserService {

	/**
	 * 返回全部列表
	 * @return
	 */
    List<TbUser> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
    PageResult findByPagination(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
    void add(TbUser user);
	
	
	/**
	 * 修改
	 */
    void update(TbUser user);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
    TbUser findById(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
    void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
    PageResult findWithConditionsByPagination(TbUser user, int pageNum, int pageSize);
	
}

File: pinyougou-parent/pinyougou-user-web/src/main/webapp/js/service/userService.js
//服务层
app.service('userService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../user/findAll.php');
    };
    //分页
    this.findByPagination = function (page, rows) {
        return $http.get('../user/findByPagination.php?page=' + page + '&rows=' + rows);
    };
    //查询实体
    this.findById = function (id) {
        return $http.get('../user/findById.php?id=' + id);
    };
    //增加
    this.add = function (entity, smsCode) {
        return $http.post('../user/add.php?smsCode=' + smsCode, entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post('../user/update.php', entity);
    };
    //删除
    this.dele = function (ids) {
        return $http.get('../user/delete.php?ids=' + ids);
    };
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../user/findWithConditionsByPagination.php?page=' + page + "&rows=" + rows, searchEntity);
    };
    //发送验证码
    this.sendCode = function (phone) {
        return $http.get("../user/sendCode.php?phone=" + phone);
    }
});
